### PR TITLE
Day5 Ex1 RecyclerView

### PR DESCRIPTION
**Pull request by at-hangtran**
Em nộp bài demo lúc sáng học và bài tập trong ngày ạ! Bài demo trong package demo còn bài mới thì trong package recyclerview :))

### CODE REVIEW

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+import static android.content.ContentValues.TAG;
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: DON'T USE IMPORT STATIC

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+class AdapterRecyclerView extends RecyclerView.Adapter {
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: Please add java doc for this class

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    private List<String> mNames = new ArrayList<>();
+
+    AdapterRecyclerView(List<String> mNames) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: Change to `AdapterRecyclerView(List<String> names) {`

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        recyclerViewHolder.mTvName.setText(mNames.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mNames.size();
+    }
+
+    private static class RecyclerViewHolder extends RecyclerView.ViewHolder {
+        final TextView mTvName;
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: add ` private `

File: app/src/main/java/demo/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        names.add("Nguyễn Văn C");
+        names.add("Nguyễn Văn D");
+        names.add("Nguyễn Văn E");
+        names.add("Nguyễn Văn F");
+        names.add("Nguyễn Văn G");
+        names.add("Nguyễn Văn K");
+        names.add("Nguyễn Văn J");
+        names.add("Nguyễn Văn D");
+        AdapterRecyclerView adapter = new AdapterRecyclerView(names);
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext());
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/MainActivity.java]: No need use `getApplicationContext()` this is a `Activity`, instead it for `this`

File: app/src/main/java/demo/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/MainActivity.java]: Please remove this line, by default is `VERTICAL`

File: app/src/main/java/recyclerview/FriendsActivity.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package recyclerview;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import vn.asiantech.internship.R;
+
+public class FriendsActivity extends AppCompatActivity {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsActivity.java]: Add java doc for this class

File: app/src/main/java/recyclerview/FriendsActivity.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main_friend);
+    }
+}
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsActivity.java]: remove this line

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+class FriendsAdapter extends RecyclerView.Adapter {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: Add java doc for this class

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    FriendsAdapter(List<User> users) {
+        this.mUsers = users;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View itemView = inflater.inflate(R.layout.item_friend_list, parent, false);
+        RecyclerView.ViewHolder v = new MyViewHolder(itemView);
+        v.setIsRecyclable(true);
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: What is this code's line ` v.setIsRecyclable(true);` mean?

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        myViewHolder.mTvName.setText(mUsers.get(position).getName());
+        myViewHolder.mTvDescription.setText(mUsers.get(position).getDescription());
+        if (position == 0 || position == 1) {
+            setBackgroundAdd(myViewHolder.mBtFriend);
+            mUsers.get(position).setState(true);
+        } else {
+            setBackgroundFriend(myViewHolder.mBtFriend);
+            mUsers.get(position).setState(false);
+        }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: remove a space line

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+                    } else {
+                        mUsers.get(getAdapterPosition()).setState(true);
+                        setBackgroundFriend(mBtFriend);
+                    }
+                }
+            });
+        }
+    }
+
+    private void setBackgroundAdd(ImageButton btn) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: change to `private void setBackgroundAdd(ImageButton imgBtnAdd) {`

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+        btn.setBackgroundResource(R.drawable.icon_friend);
+    }
+
+    private void setBackgroundFriend(ImageButton btn) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: Change to `private void setBackgroundFriend(ImageButton imgBtn...) {`

File: app/src/main/java/recyclerview/FriendsFragment.java [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+public class FriendsFragment extends Fragment {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsFragment.java]: Add java doc for this class

File: app/src/main/java/recyclerview/FriendsFragment.java [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.recyclerView);
+        List<User> users = new ArrayList<>();
+        String[] names = getResources().getStringArray(R.array.name);
+        String[] descriptions = getResources().getStringArray(R.array.description);
+        for (int i = 0; i < names.length; i++) {
+            users.add(new User(names[i], descriptions[i]));
+        }
+        FriendsAdapter adapter = new FriendsAdapter(users);
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getActivity());
+        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsFragment.java]: remove this line, by default is `VERTICAL`

File: app/src/main/java/recyclerview/User.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
+package recyclerview;
+
+class User {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/User.java]: Add java doc for this class

File: app/src/main/res/layout/activity_main.xml [hunk 17]
```diff
@@ -1,19 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/activity_main"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main.xml]: This is `RelativeLayout ` why id is `activity_main`???

File: app/src/main/res/layout/item_friend_list.xml [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+    android:layout_marginLeft="@dimen/linearlayout_margin_left_right"
+    android:layout_marginRight="@dimen/linearlayout_margin_left_right"
+    android:layout_marginTop="@dimen/linearlayout_margin_top_bottom"
+    android:background="@drawable/custom_recyclerview"
+    android:gravity="center"
+    android:orientation="horizontal"
+    android:padding="@dimen/linearlayout_padding">
+
+    <ImageView
+        android:id="@+id/imageView"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_friend_list.xml]: Change id to `img...`

File: app/src/main/res/layout/item_friend_list.xml [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+        android:gravity="center"
+        android:orientation="vertical">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="horizontal">
+
+            <TextView
+                android:id="@+id/textView"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_friend_list.xml]: change id to `tv...`

File: app/src/main/res/layout/item_friend_list.xml [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+            <TextView
+                android:id="@+id/tvDescription"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginLeft="@dimen/textview_margin_left" />
+        </LinearLayout>
+    </LinearLayout>
+
+    <ImageButton
+        android:id="@+id/btnFriend"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_friend_list.xml]: Change id to `imgBtnFriend`

File: app/src/main/res/layout/activity_main.xml (hunk 17 above)
Comment (DieuHang091095) [inline app/src/main/res/layout/activity_main.xml]: I think there is no problem with any layout :p

File: app/src/main/res/layout/activity_main.xml [hunk 21]
```diff
@@ -1,19 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout
```
Comment (DieuHang091095) [inline app/src/main/res/layout/activity_main.xml]: I don't know the problem in this layout. It is in master branch and I cloned it

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+        recyclerViewHolder.mTvName.setText(mNames.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mNames.size();
+    }
+
+    private static class RecyclerViewHolder extends RecyclerView.ViewHolder {
+        final private TextView mTvName;
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: `private final` is better!

File: app/src/main/java/demo/MainActivity.java [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+public class MainActivity extends AppCompatActivity {
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/MainActivity.java]: Add java doc for this class

File: app/src/main/res/layout/activity_main_friend.xml [hunk 24]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main_friend.xml]: Change file name to `activity_friend`

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+
+    FriendsAdapter(List<User> users) {
+        this.mUsers = users;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View itemView = inflater.inflate(R.layout.item_friend_list, parent, false);
+        RecyclerView.ViewHolder v = new MyViewHolder(itemView);
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: remove this line and `return new MyViewHolder(itemView);`

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mUsers.size();
+    }
+
+    private class MyViewHolder extends RecyclerView.ViewHolder {
+        final private TextView mTvName;
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: `private final`

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+            mUsers.get(position).setState(false);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mUsers.size();
+    }
+
+    private class MyViewHolder extends RecyclerView.ViewHolder {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: Add java doc for this class

File: app/src/main/res/layout/activity_main_friend.xml [hunk 28]
```diff
@@ -0,0 +1,12 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main_friend.xml]: remove this line `android:orientation="vertical"`

File: app/src/main/res/layout/friend_fragment.xml [hunk 29]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/friend_fragment.xml]: remove this line `android:orientation="vertical"`

File: app/src/main/res/layout/friend_fragment.xml [hunk 30]
```diff
@@ -0,0 +1,12 @@
...
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recyclerView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_gravity="center" />
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/friend_fragment.xml]: remove this line`android:layout_gravity="center"`

File: app/src/main/res/drawable/custom_recyclerview.xml [hunk 31]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/drawable/custom_recyclerview.xml]: Change file name `bg_item_list_friend`

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 32]
```diff
@@ -0,0 +1,63 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to collect and display data to the View.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: remove this break line

File: app/src/main/java/demo/MainActivity.java [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to display the recyclerview.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/MainActivity.java]: remove this break line

File: app/src/main/java/recyclerview/FriendsActivity.java [hunk 34]
```diff
@@ -0,0 +1,23 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Use to contain fragment that had a recyclerview
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsActivity.java]: remove this break line

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 35]
```diff
@@ -0,0 +1,87 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to collect and display list data to the View.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: remove this break line

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 36]
```diff
@@ -0,0 +1,63 @@
...
+        RecyclerViewHolder recyclerViewHolder = (RecyclerViewHolder) holder;
+        recyclerViewHolder.mTvName.setText(mNames.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mNames.size();
+    }
+
+    private static class RecyclerViewHolder extends RecyclerView.ViewHolder {
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: Add java doc

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 37]
```diff
@@ -0,0 +1,87 @@
...
+    @Override
+    public int getItemCount() {
+        return mUsers.size();
+    }
+
+    private class MyViewHolder extends RecyclerView.ViewHolder {
+        private final TextView mTvName;
+        private final TextView mTvDescription;
+        private final ImageButton mBtFriend;
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: Variable name `mImgBtnFriend`

File: app/src/main/java/recyclerview/FriendsFragment.java [hunk 38]
```diff
@@ -0,0 +1,44 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to display recyclerview.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsFragment.java]: remove this break line

File: app/src/main/java/recyclerview/User.java [hunk 39]
```diff
@@ -0,0 +1,45 @@
+package recyclerview;
+
+/**
+ * as a model class,get and set data to User object
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/User.java]: remove this break line

File: app/src/main/res/layout/friend_fragment.xml [hunk 40]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/friend_fragment.xml]: change file name to `fragment_friend`

File: app/src/main/res/layout/item.xml [hunk 41]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item.xml]: change file name to `item_list_...`

File: app/src/main/res/values/strings.xml [hunk 42]
```diff
@@ -1,3 +1,30 @@
 <resources>
-    <string name="app_name">Internship</string>
+    <string name="app_name">TestRecyleview</string>
+    <string name="image.description">person</string>
+    <string name="recycleview.textview.name">Name:</string>
+    <string name="recycleview.textview.description">Desciption:</string>
+    <string-array name="name">
```
Comment (at-vinhhuynh) [inline app/src/main/res/values/strings.xml]: `array-name` to `names`

File: app/src/main/res/values/strings.xml [hunk 43]
```diff
@@ -1,3 +1,30 @@
...
+        <item>Nguyễn Văn C</item>
+        <item>Nguyễn Văn D</item>
+        <item>Nguyễn Văn E</item>
+        <item>Nguyễn Văn F</item>
+        <item>Nguyễn Văn G</item>
+        <item>Nguyễn Văn H</item>
+        <item>Nguyễn Văn I</item>
+        <item>Nguyễn Văn J</item>
+    </string-array>
+    <string-array name="description">
```
Comment (at-vinhhuynh) [inline app/src/main/res/values/strings.xml]: `array-name` to `descriptions`

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 44]
```diff
@@ -0,0 +1,94 @@
...
+            setBackgroundFriend(myViewHolder.mImgBtnFriend);
+            mUsers.get(position).setState(false);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mUsers.size();
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: remove a break line

File: app/src/main/res/layout/activity_demo.xml [hunk 45]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/activity_main"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_demo.xml]: this is a `RelativeLayout` why `id` is `activity_main`?

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 46]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * Used to collect and display data to the View.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+class AdapterRecyclerView extends RecyclerView.Adapter {
+    private List<String> mNames = new ArrayList<>();
+    public static final String TAG = "Log position";
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: change `public` to `private`

File: app/src/main/java/recyclerview/FriendsActivity.java [hunk 47]
```diff
@@ -0,0 +1,22 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Use to contain fragment that had a recyclerview
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+public class FriendsActivity extends AppCompatActivity {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsActivity.java]: Change class name to `FriendActivity`

File: app/src/main/java/recyclerview/FriendsAdapter.java [hunk 48]
```diff
@@ -0,0 +1,93 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to collect and display list data to the View.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+class FriendsAdapter extends RecyclerView.Adapter {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsAdapter.java]: change class name to `FriendAdapter`

File: app/src/main/java/demo/AdapterRecyclerview.java [hunk 49]
```diff
@@ -0,0 +1,69 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to collect and display data to the View.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+class AdapterRecyclerView extends RecyclerView.Adapter {
```
Comment (at-vinhhuynh) [inline app/src/main/java/demo/AdapterRecyclerview.java]: Why class name is `AdapterRecyclerView` and file name is  `AdapterRecyclerview`? 
Please fix it!

File: app/src/main/java/recyclerview/FriendsFragment.java [hunk 50]
```diff
@@ -0,0 +1,43 @@
...
+import vn.asiantech.internship.R;
+
+/**
+ * Used to display recyclerview.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+public class FriendsFragment extends Fragment {
```
Comment (at-vinhhuynh) [inline app/src/main/java/recyclerview/FriendsFragment.java]: Change class name to `FriendFragment`

**Pull request merged**